JustSaying.UnitTests/Messaging/MessageProcessingStrategies/MessageLoopTests.cs

```diff
@@ -85,52 +106,65 @@ public async Task SimulatedListenLoop_WhenThrottlingOccurs_CallsMessageMonitor(i
         [InlineData(1, 2)]
         [InlineData(2, 2)]
         [InlineData(5, 10)]
+        [InlineData(9, 10)]
+        [InlineData(10, 10)]
         [InlineData(10, 50)]
+        [InlineData(49, 50)]
         [InlineData(50, 50)]
         public async Task SimulatedListenLoop_WhenThrottlingDoesNotOccur_DoNotCallMessageMonitor(int messageCount, int capacity)
         {
             messageCount.ShouldBeLessThanOrEqualTo(capacity,
                 "To avoid throttling, message count must be not be over capacity");
 
             var fakeMonitor = Substitute.For<IMessageMonitor>();
-            var messageProcessingStrategy = new Throttled(capacity, fakeMonitor);
+            var logger = Substitute.For<ILogger>();
+            var messageProcessingStrategy = new Throttled(capacity, Timeout.InfiniteTimeSpan, fakeMonitor, logger);
             var counter = new ThreadSafeCounter();
 
             var actions = BuildFakeIncomingMessages(messageCount, counter);
 
             await ListenLoopExecuted(actions, messageProcessingStrategy);
 
-            fakeMonitor.DidNotReceive().IncrementThrottlingStatistic();
+            if (messageCount == capacity)
```

Comment:
Can you please separate the tests where the `messageCount == capacity` into another test method so you can avoid the `if` and make what you want to assert more clear?

Reply:
Behaviour needs discussion overall, for now I just wanted it passing.

Reply:
Changed to separate the two cases in later commits.